pkg/driver/driver_aws.go

```diff
@@ -438,9 +562,86 @@ func (d *AWSDriver) GetVMs(machineID string) (VMs, error) {
 		}
 	}
 
+	err = d.getUntaggedSpotInstances(svc, machineID, clusterName, listOfVMs)
+	if err != nil {
+		return nil, err
+	}
+
 	return listOfVMs, nil
 }
 
+// getUntaggedSpotInstances appends untagged spot instances to the listOfVMs (driver.Create is non-atomic in case of spot instances)
+func (d *AWSDriver) getUntaggedSpotInstances(svc *ec2.EC2, machineID, clusterName string, listOfVMs map[string]string) error {
+	runResultSpotInstanceReqs, err := svc.DescribeSpotInstanceRequests(&ec2.DescribeSpotInstanceRequestsInput{
```

Comment:
the default DescribeSpotInstance does not paginate, you can specify maxResults to paginate the response.  https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotInstanceRequests.html

Reply:
I've removed this code completely.